contracts/transfer/tests/wrapper/alice/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+#![no_std]
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+use canonical_derive::Canon;
+use dusk_abi::{ContractId, Transaction};
+use phoenix_core::Note;
+
+pub const TX_PING: u8 = 0x01;
+pub const TX_WITHDRAW: u8 = 0x02;
+pub const TX_WITHDRAW_TO_CONTRACT: u8 = 0x03;
+
+#[derive(Debug, Clone, Canon)]
+pub struct Alice {
+    transfer: ContractId,
+}
+
+impl From<ContractId> for Alice {
+    fn from(transfer: ContractId) -> Self {
+        Self { transfer }
+    }
+}
+
+impl Alice {
+    pub const fn identifier() -> &'static [u8; 5] {
+        b"alice"
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+mod hosted {
+    use super::*;
+
+    use canonical::{Canon, CanonError, Sink, Source};
+    use dusk_abi::{ContractState, ReturnValue};
+    use rusk_abi::PaymentInfo;
+    use transfer_contract::Call;
+
+    const PAGE_SIZE: usize = 1024 * 4;
+
+    impl Alice {
+        pub fn ping(&mut self) {
+            dusk_abi::debug!("Alice ping");
```

Comment:
This is a test contract. It is testing the capability of performing a single call - the debug is useful in case of a panic that could or could not have called the contract query